File: wp-angular-starter/admin/app/services/ListingService.js
/**
 * Created by Darko on 3/10/2016.
 */

WPAngularStarter.factory("ListingService", ["$http", "apiURL", function ($http, apiURL){
    var factory = {};

    factory.deleteListing = function (id){
        return $http.delete(apiURL + "/listing/" + id);
    };

    factory.getListingsByUser = function (id){
        return $http.get(apiURL + "/listing/user/" + id);
    };

    factory.getAllReports = function (){
        return $http.get(apiURL + "/listing/report");
    };

    factory.getAllUnreadReports = function (){
        return $http.get(apiURL + "/listing/report/unread");
    };

    factory.getReport = function (id){
        return $http.get(apiURL + "/listing/report/" + id);
    };

    factory.updateReport = function (id, seen){
        var data = $.param({
            "seen": seen
        });
        var config = {headers: {'Content-Type': 'application/x-www-form-urlencoded'}};
        return $http.post(apiURL + "/listing/report/update/" + id, data, config);
    };

    factory.getGraphData = function (){
        return $http.get(apiURL + "/listing/graph");
    };

    return factory;
}]);

File: web-programming-spring-mvc-starter/src/main/java/mk/ukim/finki/wp/service/ListingService.java
package mk.ukim.finki.wp.service;

import mk.ukim.finki.wp.model.Listing;
import mk.ukim.finki.wp.model.Report;
import mk.ukim.finki.wp.model.User;
import org.springframework.web.multipart.MultipartFile;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

/**
 * Created by Darko on 2/21/2016.
 */
public interface ListingService {
    Listing createListing(String title, String content, Date createdOn, ArrayList<MultipartFile> images, User user, String locationName, String lat, String lng);

    void updateListing(Listing listing);

    void delete(Long id);

    Listing getById(Long id);

    List<Listing> getAll();

    List<Listing> getAllListingsByUser(Long userId);

    User getUser(Long userId);

    List<Listing> search(String keyword);

    List<Listing> filterByDate(Integer day, Integer month, Integer year);

    List<Report> getAllReports();

    List<Report> getAllUnreadReports();

    void saveReport(String content, User userFrom, Listing listing);

    void updateReport(Long id, Boolean seen);

    Report getReportById(Long id);

    List<Listing> nearbyListingsSearchByLocation(String currentLat, String currentLng, String maxDistance);

    List<Listing> getPiece(int offset, int end);
}
